src/uwtools/drivers/forecast.py

```diff
@@ -121,15 +138,75 @@ def requirements(self) -> None:
         ???
         """
 
-    def resources(self) -> None:
+    def resources(self, platform: dict) -> Mapping:
         """
         ???
         """
-
-    def run(self):
+        # Add required fields to platform.
+        # Currently supporting only slurm scheduler.
+        return {
+            "account": platform["account"],
+            "nodes": 1,
+            "queue": "batch",
+            "scheduler": self.job_scheduler,
+            "tasks_per_node": 1,
+            "walltime": "00:01:00",
+        }
+
+    def run(self) -> None:
         """
-        Runs FV3.
+        Runs FV3 either as a subprocess or by submitting a batch script.
         """
+        # Read in the config file.
+        forecast_config = self.config_data["forecast"]
+        platform_config = self.config_data["platform"]
+
+        # Prepare directories.
+        run_directory = forecast_config["RUN_DIRECTORY"]
+        self.create_directory_structure(run_directory, "rename")
+
+        static_files = forecast_config["STATIC"]
+        self.stage_files(run_directory, static_files, link_files=True)
+        cycledep_files = forecast_config["CYCLEDEP"]
+        self.stage_files(run_directory, cycledep_files, link_files=True)
+
+        # Create the job script.
+        platform_resources = self.resources(platform_config)
+        batch_script = self.batch_script(platform_resources)
+        args = "--export=None"
+        run_command = self.run_cmd(
+            args,
+            run_cmd=platform_config["MPICMD"],
+            exec_name=forecast_config["EXEC_NAME"],
+        )
+        batch_script.append(run_command)
+
+        if self._dry_run:
+            # Apply switch to allow user to view the run command of config.
+            # This will not run the job.
+            logging.info("Batch Script:")
+            logging.info(batch_script)
+            return
+
+        # Run the job.
+        if self._batch_script is not None:  # pragma: no cover
+            outpath = run_directory.join(self._batch_script)
```

Comment:
What is `outpath` set to here? If I try this in the REPL I get
```
>>> "/some/dir".join("a-file")
'a/some/dir-/some/dirf/some/diri/some/dirl/some/dire'
```
Unless I'm missing something, it seems like this would have caused problems in a unit test -- but maybe the `# pragma: no cover` is masking that, so it would be good to remove that.

Reply:
Interesting - thanks for catching that. I was thinking about how to catch the true run in a unit test without it actually running, which is why I had been leaving it out

Reply:
There are a couple small gaps in my calendar today -- send me an invitation if you'd like to pair on this.

Reply:
replaced with `Path(run_directory) / self._batch_script`